source/usage-examples/findOne.txt

```diff
@@ -3,3 +3,99 @@ Find a Document
 ===============
 
 .. default-domain:: mongodb
+
+You can retrieve a single document from your collection with the
+``Collection.FindOne()`` method. 
+
+..
+   method
+
+The ``FindOne()`` method takes a filter document. Calling the ``FindOne()``
+method retrieves a document from your MongoDB instance that matches your
+filter document. If you pass the ``FindOne()`` method an empty filter document,
+MongoDB considers your entire collection when retrieving your document. For more
+information on documents in the MongoDB Go driver, see our
+:doc:`guide on data formats </fundamentals/data-formats>`.
+
+You must pass the ``FindOne()`` method a ``context.Context`` interface type
+value. For more information on the ``context.Context`` interface and the
+``context`` package, see our :doc:`guide on context </fundamentals/context>`.
+
+..
+   options
+
+You can specify options that change the behavior of the ``FindOne()`` method.
+One example of an option is **sort**, which instructs your MongoDB instance to
+order the matching documents of your collection before it retrieves the
+first document. Another example of an option is a **projection**, which
+specifies or restricts the fields in your returned document. You can specify
+options by passing the ``FindOne()`` method a pointer to a ``FindOneOptions`` struct.  
+You can configure a ``FindOneOptions`` struct by using setter methods such as the
+``FindOneOptions.SetProjection()`` method. For more information on the sort operation,
+:doc:`see our guide on sorting </fundamentals/crud/read-operations/sort/>`.
+For more information on the projection operation, 
+:doc:`see our guide on projections </fundamentals/crud/read-operations/project/>`.  
+
+..
+   return
+
+If the query filter you specified in your ``FindOne()`` method call matches one
+or more documents in your collection, it returns the first matched result. The
+return type ``SingleResult`` represents a single document returned from a query
+to a MongoDB instance. You can call the ``SingleResult.Decode()`` method
+to unmarshal a ``SingleResult`` struct type into a variable. Pass the ``Decode()`` 
+method a pointer to the variable that you would like to contain your returned
+document. 
+
+..
+   error 
+
+When you call the ``Decode()`` method to unmarshal your result, the method
+returns an ``error`` interface type which can contain one of the following
+values: 
+
+- ``nil`` if a document matched your query, and there were no errors retrieving
+  the document. 
+- ``ErrNoDocuments`` if no documents matched your query. 
+- If there was an error retrieving your document during execution of the
```

Comment:
It will also return an error if there was an error unmarshalling your result

Reply:
Thanks for the note. To confirm, Is the correct error `[InvalidUnmarshalError](https://github.com/mongodb/mongo-go-driver/blob/55b507b3e45a2ef5dc1d5a4cba3333c45dce443c/bson/unmarshal.go#L35)`?

Fix: https://github.com/mongodb/docs-golang/pull/2/commits/9a28730a683d23a022fb3d832ad9f7e3ed6892e3

Reply:
I wouldn't specify the error or that it's a bson.D, because Decode takes non-bson.D values too and those can have different errors. Maybe something like this at the end of the list:
`if the document was successfully retrieved, but unmarshaling failed, Decode() will return the unmarshaling error`

Reply:
Thanks for the note. I have taken your suggested edit and tweaked it to fit in line with our [style guide](https://docs.mongodb.com/meta/style-guide/).

Fix: https://github.com/mongodb/docs-golang/pull/2/commits/bf417a7b18cbb5c9aa7f78508a519c1123cc8897

For my own edification, is there a reason why you did not want a specific error mentioned? Are there a few possible errors that could occur during unmarshaling?

Reply:
Looks good! Yeah, depending on few different factors (like what you pass in, and what registry you're unmarshalling with) the errors can be different.